src/events/components/AttendButton/AttendButton.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { FC, useState, useEffect } from 'react';
+import Button from 'core/components/errors/NotAuthenticated/Button';
+import CaptchaModal from './CaptchaModal';
+import { DateTime } from 'luxon';
+import { getAttendeeForEvent, userAttendEvent } from 'events/api/attendee';
+
+interface IAttendButtonProps {
+  eventId: number;
+  registrationStart: DateTime;
+  registrationEnd: DateTime;
+  unattendDeadline: DateTime;
+}
+
+const AttendButton: FC<IAttendButtonProps> = (props: IAttendButtonProps) => {
+  const { eventId, registrationEnd, registrationStart, unattendDeadline } = props;
+  const [showModal, setShowModal] = useState<boolean>(false);
+  const [attending, setAttending] = useState<boolean>(false); // Fetch is attending
+  const currentTime = DateTime.local(); // Check if this works
+
+  useEffect(() => {
+    const fetchAttending = async () => {
+      const attendingFetch = await getAttendeeForEvent(eventId);
+      setAttending(attendingFetch.attended);
+    };
+    fetchAttending();
+  }, [eventId]);
```

Comment:
Not sure if we need eventId on the useEffect. It should fetch only once anyway since you need to change url to go to another event?

Reply:
Changing the URL is what triggers the eventId to change though.
If you were to go from one event page to another, I'm you would still have the data from the first event page you visited.
This is why all dependencies should always be in the array.